storage/innobase/handler/ha_innodb.cc

```diff
@@ -12923,7 +12923,7 @@ static bool innobase_ddse_dict_init(
     return true;
   }
 
-  if (!innobase_is_ddse()) {
+  if (!innobase_is_ddse() && !is_dd_engine_change) {
```

Comment:
> Some of the previous PRs used a different check here, i.e. ha_thd()->is_dd_system_thread(). Would it work here?

I think it won't work. since in both case(innodb ddse, myrocks ddse), it will call these ddse methods in is_dd_system_thread().. 

Let me also replace is_dd_system_thread() with the new method

Reply:
also disable rocksdb ddse combination first..

Reply:
@laurynas-biveinis , still see some failures such as rocksdb.tmp_table mtr in myrocks DDSE
still investigating